src/Core/Framework/DataAbstractionLayer/Dbal/Common/RepositoryIterator.php

```diff
@@ -71,19 +90,29 @@ public function getTotal(): int
      */
     public function fetchIds(): ?array
     {
+        if ($this->endReached) {
+            return null;
```

Comment:
we could make call rewind/reset here and don't leak it as public api. So it would be something completly internal